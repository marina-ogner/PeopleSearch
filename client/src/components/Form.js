import React from "react";

export default class Form extends React.Component {
    state = {
        firstName: "",
        firstNameError: "",
        lastName: "",
        lastNameError: "",
        state: "",
        stateError: "",
    };

    change = e => {
        this.setState({
            [e.target.name]: e.target.value
        });
    };

    validate = () => {
        let isError = false;
        const errors = {
            firstNameError: "",
            lastNameError: "",
            stateError: ""
        };
        if (this.state.firstName.trim().length < 1) {
            isError = true;
            errors.firstNameError = "Please enter First Name";
        }
        if (this.state.lastName.trim().length < 1) {
            isError = true;
            errors.lastNameError = "Please enter Last Name";
        }
        if (this.state.state.trim().length < 1) {
            isError = true;
            errors.stateError = "Please enter State";
        }
        this.setState({
            ...this.state,
            ...errors
        });
        return isError;
    };

    onSubmit = e => {
        e.preventDefault();
        const err = this.validate();
        if (!err) {
            // clear form
            this.setState({
                firstName: "",
                firstNameError: "",
                lastName: "",
                lastNameError: "",
                state: "",
                stateError: "",
            });
        }
        this.props.getPeople(e);
    };
    render() {
        return (
            <form className="form-inline needs-validation" onSubmit={e => this.onSubmit(e)}>
                <div className="form-group top">
                    <label className="sm-0">First Name </label>
                    <input
                        value={this.state.firstName}
                        onChange={e => this.change(e)}
                        disabled={this.props.load === true}
                        type="text"
                        className="form-control"
                        name="firstName"
                        placeholder="Elon" />
                    <div className="invalid-feedback inputError">
                        {this.state.firstNameError}
                    </div>
                </div>
                <div className="form-group">
                    <label className="sm-0">Last Name</label>
                    <input
                        value={this.state.lastName}
                        onChange={e => this.change(e)}
                        disabled={this.props.load === true}
                        type="text"
                        className="form-control"
                        name="lastName"
                        placeholder="Musk" />
                    <div className="invalid-feedback inputError">
                        {this.state.lastNameError}
                    </div>
                </div>
                <div className="form-group">
                    <label className="sm-0">State</label>
                    <input
                        value={this.state.state}
                        onChange={e => this.change(e)}
                        disabled={this.props.load === true}
                        type="text"
                        className="form-control"
                        name="state"
                        placeholder="LA" />
                    <div className="invalid-feedback inputError">
                        {this.state.stateError}
                    </div>
                </div>
                <button
                    disabled={this.props.load === true}
                    id="button"
                    className="btn btn-outline-primary"> Submit
                </button>
            </form>
        );
    }
};