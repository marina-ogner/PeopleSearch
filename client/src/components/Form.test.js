import React from 'react';
import Form from './Form';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

test('Form component should render as expected', () => {
    const component = shallow(<Form />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
})
test('Form component validation with a missing input should keep entered values and create an error', () => {
    const getPeople = async (e) => { };
    const component = mount(<Form getPeople={getPeople} />);
    const inp = component.find('input');
    inp.at(0).simulate("change", { target: { value: "MARK", name: "firstName" } });
    inp.at(1).simulate("change", { target: { value: "PRENTICE", name: "lastName" } });
    inp.at(2).simulate("change", { target: { value: "", name: "state" } });
    const form = component.find('form');
    form.simulate('submit', { preventDefault: () => { } });
    expect(component.state().firstName.length).not.toBe(0);
    expect(component.state().lastName.length).not.toBe(0);
    expect(component.state().state.length).toBe(0);
    expect(component.state().firstNameError.length).toBe(0);
    expect(component.state().lastNameError.length).toBe(0);
    expect(component.state().stateError.length).not.toBe(0);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
})
test('Form component validation with two missing inputs should keep entered value and create errors', () => {
    const getPeople = async (e) => { };
    const component = mount(<Form getPeople={getPeople} />);
    const inp = component.find('input');
    inp.at(0).simulate("change", { target: { value: "", name: "firstName" } });
    inp.at(1).simulate("change", { target: { value: "", name: "lastName" } });
    inp.at(2).simulate("change", { target: { value: "WA", name: "state" } });
    const form = component.find('form');
    form.simulate('submit', { preventDefault: () => { } });
    expect(component.state().firstName.length).toBe(0);
    expect(component.state().lastName.length).toBe(0);
    expect(component.state().state.length).not.toBe(0);
    expect(component.state().firstNameError.length).not.toBe(0);
    expect(component.state().lastNameError.length).not.toBe(0);
    expect(component.state().stateError.length).toBe(0);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
})
test('Form component validation with all missing inputs', () => {
    const getPeople = async (e) => { };
    const component = mount(<Form getPeople={getPeople} />);
    const inp = component.find('input');
    inp.at(0).simulate("change", { target: { value: "", name: "firstName" } });
    inp.at(1).simulate("change", { target: { value: "", name: "lastName" } });
    inp.at(2).simulate("change", { target: { value: "", name: "state" } });
    const form = component.find('form');
    form.simulate('submit', { preventDefault: () => { } });
    expect(component.state().firstName.length).toBe(0);
    expect(component.state().lastName.length).toBe(0);
    expect(component.state().state.length).toBe(0);
    expect(component.state().firstNameError.length).not.toBe(0);
    expect(component.state().lastNameError.length).not.toBe(0);
    expect(component.state().stateError.length).not.toBe(0);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
})
test('Form component validation with all entered inputs', () => {
    const getPeople = async (e) => { };
    const component = mount(<Form getPeople={getPeople} />);
    const inp = component.find('input');
    inp.at(0).simulate("change", { target: { value: "MARK", name: "firstName" } });
    inp.at(1).simulate("change", { target: { value: "PRENTICE", name: "lastName" } });
    inp.at(2).simulate("change", { target: { value: "WA", name: "state" } });
    const form = component.find('form');
    form.simulate('submit', { preventDefault: () => { } });
    expect(component.state().firstName.length).toBe(0);
    expect(component.state().lastName.length).toBe(0);
    expect(component.state().state.length).toBe(0);
    expect(component.state().firstNameError.length).toBe(0);
    expect(component.state().lastNameError.length).toBe(0);
    expect(component.state().stateError.length).toBe(0);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
})
