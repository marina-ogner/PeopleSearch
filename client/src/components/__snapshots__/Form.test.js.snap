// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render as expected 1`] = `
<form
  className="form-inline needs-validation"
  onSubmit={[Function]}
>
  <div
    className="form-group top"
  >
    <label
      className="sm-0"
    >
      First Name 
    </label>
    <input
      className="form-control"
      disabled={false}
      name="firstName"
      onChange={[Function]}
      placeholder="Elon"
      type="text"
      value=""
    />
    <div
      className="invalid-feedback inputError"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="sm-0"
    >
      Last Name
    </label>
    <input
      className="form-control"
      disabled={false}
      name="lastName"
      onChange={[Function]}
      placeholder="Musk"
      type="text"
      value=""
    />
    <div
      className="invalid-feedback inputError"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="sm-0"
    >
      State
    </label>
    <input
      className="form-control"
      disabled={false}
      name="state"
      onChange={[Function]}
      placeholder="LA"
      type="text"
      value=""
    />
    <div
      className="invalid-feedback inputError"
    />
  </div>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    id="button"
  >
     Submit
  </button>
</form>
`;

exports[`Form component validation with a missing input should keep entered values and create an error 1`] = `
<Form
  getPeople={[Function]}
>
  <form
    className="form-inline needs-validation"
    onSubmit={[Function]}
  >
    <div
      className="form-group top"
    >
      <label
        className="sm-0"
      >
        First Name 
      </label>
      <input
        className="form-control"
        disabled={false}
        name="firstName"
        onChange={[Function]}
        placeholder="Elon"
        type="text"
        value="MARK"
      />
      <div
        className="invalid-feedback inputError"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        Last Name
      </label>
      <input
        className="form-control"
        disabled={false}
        name="lastName"
        onChange={[Function]}
        placeholder="Musk"
        type="text"
        value="PRENTICE"
      />
      <div
        className="invalid-feedback inputError"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        State
      </label>
      <input
        className="form-control"
        disabled={false}
        name="state"
        onChange={[Function]}
        placeholder="LA"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      >
        Please enter State
      </div>
    </div>
    <button
      className="btn btn-outline-primary"
      disabled={false}
      id="button"
    >
       Submit
    </button>
  </form>
</Form>
`;

exports[`Form component validation with all entered inputs 1`] = `
<Form
  getPeople={[Function]}
>
  <form
    className="form-inline needs-validation"
    onSubmit={[Function]}
  >
    <div
      className="form-group top"
    >
      <label
        className="sm-0"
      >
        First Name 
      </label>
      <input
        className="form-control"
        disabled={false}
        name="firstName"
        onChange={[Function]}
        placeholder="Elon"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        Last Name
      </label>
      <input
        className="form-control"
        disabled={false}
        name="lastName"
        onChange={[Function]}
        placeholder="Musk"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        State
      </label>
      <input
        className="form-control"
        disabled={false}
        name="state"
        onChange={[Function]}
        placeholder="LA"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      />
    </div>
    <button
      className="btn btn-outline-primary"
      disabled={false}
      id="button"
    >
       Submit
    </button>
  </form>
</Form>
`;

exports[`Form component validation with all missing inputs 1`] = `
<Form
  getPeople={[Function]}
>
  <form
    className="form-inline needs-validation"
    onSubmit={[Function]}
  >
    <div
      className="form-group top"
    >
      <label
        className="sm-0"
      >
        First Name 
      </label>
      <input
        className="form-control"
        disabled={false}
        name="firstName"
        onChange={[Function]}
        placeholder="Elon"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      >
        Please enter First Name
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        Last Name
      </label>
      <input
        className="form-control"
        disabled={false}
        name="lastName"
        onChange={[Function]}
        placeholder="Musk"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      >
        Please enter Last Name
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        State
      </label>
      <input
        className="form-control"
        disabled={false}
        name="state"
        onChange={[Function]}
        placeholder="LA"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      >
        Please enter State
      </div>
    </div>
    <button
      className="btn btn-outline-primary"
      disabled={false}
      id="button"
    >
       Submit
    </button>
  </form>
</Form>
`;

exports[`Form component validation with two missing inputs should keep entered value and create errors 1`] = `
<Form
  getPeople={[Function]}
>
  <form
    className="form-inline needs-validation"
    onSubmit={[Function]}
  >
    <div
      className="form-group top"
    >
      <label
        className="sm-0"
      >
        First Name 
      </label>
      <input
        className="form-control"
        disabled={false}
        name="firstName"
        onChange={[Function]}
        placeholder="Elon"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      >
        Please enter First Name
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        Last Name
      </label>
      <input
        className="form-control"
        disabled={false}
        name="lastName"
        onChange={[Function]}
        placeholder="Musk"
        type="text"
        value=""
      />
      <div
        className="invalid-feedback inputError"
      >
        Please enter Last Name
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="sm-0"
      >
        State
      </label>
      <input
        className="form-control"
        disabled={false}
        name="state"
        onChange={[Function]}
        placeholder="LA"
        type="text"
        value="WA"
      />
      <div
        className="invalid-feedback inputError"
      />
    </div>
    <button
      className="btn btn-outline-primary"
      disabled={false}
      id="button"
    >
       Submit
    </button>
  </form>
</Form>
`;
